src/control/lib/support/log.go

```diff
@@ -659,85 +669,232 @@ func copyServerConfig(log logging.Logger, opts ...CollectLogsParams) error {
 	return nil
 }
 
-// Copy only specific lines from the server logs based on the Start/End date
-// and Start/End time, provided by user.
-func cpLinesFromLog(log logging.Logger, srcFile string, destFile string, opts ...CollectLogsParams) error {
-	// Copy the full log file in case of no dates provided
-	if opts[0].LogStartDate == "" && opts[0].LogEndDate == "" {
-		err := cpLogFile(srcFile, destFile, log)
-		if err != nil {
-			return err
-		}
-	} else {
-		var startTime, endTime string
+// Calculate the start/end time provided by user.
+func getDateTime(log logging.Logger, opts ...CollectLogsParams) (time.Time, time.Time, error) {
+	var startTime, endTime, dateTimeFormat string
+	currentTime := time.Now()
 
-		logStartDate, err := time.Parse(MMDD, opts[0].LogStartDate)
-		if err != nil {
-			return err
-		}
+	logStartDate, err := time.Parse(MMDD, opts[0].LogStartDate)
+	if err != nil {
+		return currentTime, currentTime, err
+	}
 
-		logEndDate, err := time.Parse(MMDD, opts[0].LogEndDate)
-		if err != nil {
-			return err
-		}
+	logEndDate, err := time.Parse(MMDD, opts[0].LogEndDate)
+	if err != nil {
+		return currentTime, currentTime, err
+	}
 
-		// Default Start time, in case no start time provides on start dates.This will copy log start of the day.
-		if opts[0].LogStartTime == "" {
-			opts[0].LogStartTime = "00:00:00"
-		}
+	// Default Start time, in case no start time provides on start dates.This will copy log start of the day.
+	if opts[0].LogStartTime == "" {
+		opts[0].LogStartTime = "00:00:00"
+	}
 
-		// Default End time, in case no End time provides.This will copy log till the End of the day.
-		if opts[0].LogEndTime == "" {
-			opts[0].LogEndTime = "23:59:59"
-		}
+	// Default End time, in case no End time provides.This will copy log till the End of the day.
+	if opts[0].LogEndTime == "" {
+		opts[0].LogEndTime = "23:59:59"
+	}
 
+	// Set the date/time format for Engine log.
+	if opts[0].LogCmd == "EngineLog" {
 		startTime = fmt.Sprintf("%d/%d-%s", logStartDate.Month(), logStartDate.Day(), opts[0].LogStartTime)
 		endTime = fmt.Sprintf("%d/%d-%s", logEndDate.Month(), logEndDate.Day(), opts[0].LogEndTime)
+		dateTimeFormat = MMDDHHMMSS
+	}
 
-		actStartTime, _ := time.Parse(MMDDHHMMSS, startTime)
-		actEndTime, _ := time.Parse(MMDDHHMMSS, endTime)
+	// Set the date/time format for Control and Helper log.
+	if opts[0].LogCmd == "ControlLog" || opts[0].LogCmd == "HelperLog" {
+		startTime = fmt.Sprintf("%d/%d %s", logStartDate.Month(), logStartDate.Day(), opts[0].LogStartTime)
+		endTime = fmt.Sprintf("%d/%d %s", logEndDate.Month(), logEndDate.Day(), opts[0].LogEndTime)
+		dateTimeFormat = MMDD_HHMMSS
+	}
```

Comment:
Maybe I am missing something, but this is a bit confusing to me. I don't understand why you must format the strings in a particular way here, when the string isn't being returned. You can use the `time.Date()` function to create a `time.Time` that equals a particular date and time, without converting to/from strings: https://pkg.go.dev/time#Date

Reply:
We have different format in different files so to make the proper expected format I used string conversion and later use the time.Parse() as return.

Helper Log/Control Log:
2023/10/17 19:06:50

Engine Logs:
10/17-19:12:32.45

Reply:
The confusion for me is that these formatted strings aren't being used anywhere. They're just being converted back into `time.Time`. If we only need a `time.Time` that represents the times that the user requested, we can do that without any formatting at all.

Reply:
It's used to create the specific time.Parse() format which is different for different logs. I am not sure about any other logic to handle both the situation without formatting.
  
MMDDHHMMSS  = "1/2-15:04:05"
MMDD_HHMMSS = "1/2 15:04:05"

Reply:
I understand there are different date formats, but the string generated here is not being compared to log date strings. `time.Time` structs are format-independent, and the user's time/date input formats aren't expected to be different between log types either. It should be fine to just parse each time/date once:

```golang
// combine the user's requested date and time strings
startTimeStr := fmt.Sprintf("%s %s", opts[0].LogStartDate, opts[0].LogStartTime)
// parse into a time.Time
startTime, err := time.ParseInLocation(MMDD_HHMMSS, startTimeStr, time.Local)
```

Reply:
Got it! updated the code and passing the time/date once.